src/main/flight/pid.c

```diff
@@ -522,9 +526,14 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
             gyroRateFiltered = dtermLowpassApplyFn((filter_t *) &dtermLowpass[axis], gyroRateFiltered);
             gyroRateFiltered = dtermLowpass2ApplyFn((filter_t *) &dtermLowpass2[axis], gyroRateFiltered);
 
-            const float rD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * currentPidSetpoint - gyroRateFiltered;    // cr - y
+            // no transition if relaxFactor == 0
+            float transition = 1;
```

Comment:
In the line 'float transition = 1;', why isn't there an 'f' after the number, like the more common 'float transition = 1.0f'

Reply:
Because it's not necessary and adds visual noise.

Reply:
OK thanks for that, I've never appreciated when it is, vs when it isn't, necessary to add the f.  Is it only if you've a number with decimal points like nnn.nn that it needs to be nnn.nnf when declaring a float?

Thanks for the rewrite, I'm very grateful.

Reply:
Basically yes, only if there's a decimal part, and if you want it to be float rather than double. For integer numbers they get converted to float automatically when involved in calculations with floats

Reply:
@DieHertz : Double math needs to be avoided to .. -Wdoublepromotion catches most cases ,,,

Reply:
There are no doubled in this code AFAIJ

Reply:
Problem is that numeric constant is treated as `double` and C requires double promotion when one argument is `double`. Constant without `f` suffix  may cause conversion to double and back to float:
```
float a,b;
b = a + 1.0;
```

Reply:
`1` has type of `int`, at least in C++. `1.0` has typ le `double`.
`1` gets an implicit conversion from int to float in my code

Reply:
Sorry, I was answering @ctzsnooze question, after reading it wrong ...

Reply:
Thanks to you both that clears up my understanding a lot.